lib/physical/package.rb

```diff
@@ -28,6 +28,14 @@ def weight
       container.weight + items_weight + void_fill_weight
     end
 
+    # Cost is optional. We will only return an aggregate if all items
+    # have cost defined. Otherwise we will retun nil.
+    # @return Money
+    def items_value
+      items_cost = items.map(&:cost)
+      items_cost.reduce(&:+) if items_cost.compact.size == items_cost.size
```

Comment:
If cost weren't optional I would have raised an error. If you ask for cost then you should know if you left it out without any ambiguity... Following that line of thought I'd make Item.cost raise an error. That isn't backwards compatible so I went with the nil. Let me know your thoughts on my alternative thoughts.

Reply:
I think the way you handled this is fine.